app/controllers/staff/reports_state_controller.rb

```diff
@@ -47,20 +51,23 @@ def update
   private def change_report_state_to(state)
     policy_action = STATE_TO_POLICY_ACTION.fetch(state)
 
-    authorize report, policy_action + "?"
+    unless report.valid?
+      authorize report
+      flash[:error] = t("action.report.#{policy_action}.failure")
+      return redirect_to report_path(report)
+    end
 
-    if report.valid?
+    unless report.state == state
+      authorize report, policy_action + "?"
       report.update!(state: state)
       report.create_activity key: "report.state.changed_to.#{state}", owner: current_user
-      @report_presenter = ReportPresenter.new(report)
-      render "staff/reports_state/#{policy_action}/complete"
-    else
-      flash[:error] = t("action.report.#{policy_action}.failure")
-      redirect_to report_path(report)
     end
+
+    @report_presenter = ReportPresenter.new(report)
+    render "staff/reports_state/#{policy_action}/complete"
   end
 
   private def report
-    @report ||= Report.find(params[:report_id])
+    @report ||= Report.lock.find(params[:report_id])
```

Comment:
What led you to decide to lock this record, because you expect it to be changed?